src/loaders/lottie/tvgLottieParser.cpp

```diff
@@ -857,9 +861,9 @@ LottieRepeater* LottieParser::parseRepeater()
 }
 
 
-LottieObject* LottieParser::parseObject()
+LottieObject* LottieParser::parseObject(const char* type)
 {
-    auto type = getString();
+    if (!type) type = getString();
     if (!type) return nullptr;
```

Comment:
In case `type` isn't nullptr from the beginning, we might be able to avoid unnecessary condition check by this code

```cpp
if (!type) {
    type = getString();
    if (!type) return nullptr;
}
```